staging/kos/pkg/controllers/subnet/validator.go

```diff
@@ -60,57 +58,8 @@ const (
 	metricsSubsystem = "subnet_validator"
 )
 
-// conflictsCache holds information for one subnet regarding conflicts with
-// other subnets. There's no guarantee that the cache is up-to-date. For
-// instance, if a conflicting  subnet X is in it and gets deleted it is not
-// removed.
-type conflictsCache struct {
-	// ownerUID is the UID of the subnet owning the conflicts cache.
-	ownerUID k8stypes.UID
-
-	// rivals approximately identifies the subnets that have a conflict with the
-	// subnet owning the cache. If subnet X is the owner of the conflicts cache
-	// and a namespaced name Y is in rivals this means that in the past a subnet
-	// with namespaced name Y was processed by a queue worker and a conflict
-	// with X was found.
-	rivals []k8stypes.NamespacedName
-}
-
-// Validator validates subnets and writes the outcome in their status.
-// Validation consists of two checks:
-//
-// 		(1) CIDRs for subnets with the same VNI are disjoint.
-// 		(2) all subnets with the same VNI are within the same K8s namespace.
-//
-// If a subnet S1 does not pass validation because of a conflict with another
-// subnet S2, upon deletion of S2 S1 is validated again. Validator uses an
-// informer to watch for subnets, but does a live list against the API server to
-// retrieve the conflicting subnets when validating a subnet. This avoids race
-// conditions caused by multiple validators running at the same time.
-type Validator struct {
-	netIfc         kosclientv1a1.NetworkV1alpha1Interface
-	subnetInformer k8scache.SharedInformer
-	subnetLister   netlistv1a1.SubnetLister
-	eventRecorder  k8seventrecord.EventRecorder
-	queue          k8sworkqueue.RateLimitingInterface
-	workers        int
-
-	// conflicts associates a subnet namespaced name with its conflictsCache.
-	// Always access while holding conflictsMutex.
-	conflicts      map[k8stypes.NamespacedName]*conflictsCache
-	conflictsMutex sync.Mutex
-
-	// staleRVs associates a subnet X's namespaced name for which there was a
-	// successful status update to X's resource version prior to the update.
-	// When a worker begins processing a subnet X, it checks whether X's
-	// resource version matches the resource version in staleRVs[X]. If that's
-	// the case X is stale, i.e. it does not reflect the latest update yet,
-	// hence processing is postponed to when the notification for the update is
-	// received.
-	// Only access while holding staleRVsMutex.
-	staleRVs      map[k8stypes.NamespacedName]string
-	staleRVsMutex sync.RWMutex
-
+// Prometheus metrics.
+var (
```

Comment:
Some day we may want to test with fake Prometheus instrumentation (https://github.com/prometheus/client_golang/blob/master/prometheus/testutil/testutil.go#L17-L24).
But today is not that day.